luigi_pipeline/lib/model/gcnv_mt_schema.py

```diff
@@ -0,0 +1,212 @@
+        return variant_id.get_expr_for_xpos(
+            hl.locus(self.mt.contig, self.mt.end)
+        )
+
+class SeqrGCNVGenotypesSchema(SeqrGenotypesSchema):
+
+    def __init__(self, *args, is_new_joint_call=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_new_joint_call = is_new_joint_call
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples(self):
+        return self._genotype_filter_samples(lambda g: True)
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_new_call(self):
+        return self._genotype_filter_samples(lambda g: g.new_call)
+
+    def samples_no_call(self):
+        pass
+
+    def samples_num_alt(self):
+        pass
+
+    def samples_gq(self):
+        pass
+
+    def samples_ab(self):
+        pass
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_qs(self, start=0, end=1000, step=10):
+        return hl.struct(**{
+            '%i_to_%i' % (i, i+step): self._genotype_filter_samples(lambda g: ((g.qs >= i) & (g.qs < i+step)))
+            for i in range(start, end, step)
+        }, **{
+            "samples_qs_gt_1000": self._genotype_filter_samples(lambda g: g.qs > 1000)
+        })
+
+    @row_annotation(name="samples_cn", fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_cn(self, start=0, end=4, step=1):
+        return hl.struct(**{
+            '%i' % i: self._genotype_filter_samples(lambda g: g.cn == i)
+            for i in range(start, end, step)
+        }, **{
+            "samples_cn_gte_4": self._genotype_filter_samples(lambda g: g.cn >= 4)
+        })
+
+    def _genotype_filter_samples(self, filter):
+        samples = self.mt.genotypes.filter(filter).map(lambda g: g.sample_id)
+        return hl.if_else(hl.len(samples) > 0, samples, hl.missing(hl.dtype('array<str>')))
+    
+    def _genotype_fields(self):
+        return {
+            'sample_id': self.mt.sample_fix.first_match_in(SAMPLE_ID_REGEX)[0],
+            'qs': self.mt.QS,
+            'cn': self.mt.CN,
+            'defragged': self.mt.defragmented,
+            # Hail expression is to bool-ify a string value.
+            'prev_call': hl.if_else(hl.len(self.mt.identical_ovl) > 0, True, False) if self._is_new_joint_call else ~self.mt.is_latest,
+            'prev_overlap': hl.if_else(hl.len(self.mt.any_ovl) > 0, True, False)  if self._is_new_joint_call else False,
+            # NB: previous implementation also falsified NA, but hail treats NA as an empty value.
+            'new_call': self.mt.no_ovl if self._is_new_joint_call else False,
+        }
+
+class SeqrGCNVVariantsAndGenotypesSchema(SeqrGCNVVariantSchema, SeqrGCNVGenotypesSchema):
+    
+    # NB: we override this method because the row keys are different.
+    @staticmethod
+    def elasticsearch_row(ds):
+        if isinstance(ds, hl.MatrixTable):
+            ds = ds.rows()
+        return table.drop(table.variant_name, table.svtype)
```

Comment:
you should be able to get `row_key = table.row_key` before you do the flattening and then do `table = table.drop(*row_key)` here, and that should work for both schemas and then you could just update the base class method

Reply:
Awesome.  Great suggestion!